earth2studio/data/cmip6.py

```diff
@@ -0,0 +1,420 @@
+        variable: str | list[str] | VariableArray,
+    ) -> xr.DataArray:
+        """Function to get data.
+
+        Parameters
+        ----------
+        time : datetime | list[datetime] | TimeArray
+            Timestamps to return data for.
+        variable : str | list[str] | VariableArray
+            Strings or list of strings that refer to variables to return.
+
+        Returns
+        -------
+        xr.DataArray
+            Loaded data array
+        """
+
+        # Prepare data inputs
+        time, variable = prep_data_inputs(time, variable)
+
+        # Convert variable to
+        var_set: set[str] = {CMIP6Lexicon[v][0][0] for v in variable}
+        cmip6_variable_id = list(var_set)
+
+        # Search for data
+        try:
+            self.catalog.search(
+                experiment_id=self.experiment_id,
+                source_id=self.source_id,
+                table_id=self.table_id,
+                variable_id=cmip6_variable_id,
+                variant_label=self.variant_label,
+                file_start=self.file_start,
+                file_end=self.file_end,
+            )
+            dsd = self.catalog.to_dataset_dict(
+                prefer_streaming=False, add_measures=False
+            )  # NOTE: it may be better to use streaming however this resulted in lots of errors
+        except Exception as e:
+            raise ValueError(
+                f"Error searching for CMIP6 data: {e}"
+                f"\nExperiment ID: {self.experiment_id}"
+                f"\nSource ID: {self.source_id}"
+                f"\nTable ID: {self.table_id}"
+                f"\nVariant Label: {self.variant_label}"
+                f"\nFile Start: {self.file_start}"
+                f"\nFile End: {self.file_end}"
+            ) from e
+
+        # Assert that we have all the data and no extra dimensions
+        dsd_keys: set[str] = set(dsd.keys())
+        cmip6_ids_set = var_set
+        if dsd_keys - cmip6_ids_set:  # pragma: no cover
+            raise IndexError(
+                f"Variable(s) {cmip6_ids_set - dsd_keys} not found in CMIP6 dataset"
+            )
+        if cmip6_ids_set - dsd_keys:  # pragma: no cover
+            raise IndexError(
+                f"Variable(s) {dsd_keys - cmip6_ids_set} not found in CMIP6 dataset"
+            )
+
+        # Get lat/lon and calendar from first dataset
+        ds = dsd[list(dsd.keys())[0]]
+
+        # Regular lat-lon grid → lat/lon 1-D coordinates exist
+        if {"lat", "lon"} <= set(ds.coords):
+            lat_1d = ds["lat"].values
+            lon_1d = ds["lon"].values
+            grid_shape = (len(lat_1d), len(lon_1d))
+            da_dims_xy = ("lat", "lon")
+            coord_dict: dict[str, object] = {
+                "lat": lat_1d,
+                "lon": lon_1d,
+            }
+        # Curvilinear ocean/ice grid → 2-D latitude/longitude coordinates
+        elif {"latitude", "longitude"} <= set(ds.coords):
```

Comment:
This section is untested in the coverage report, do we know of datasets that are like this?

Reply:
Hmm, thats weird. The SST uses this curvlinear grid and that is in the unit tests...